### PR TITLE
Login auth

### PR DESCRIPTION
**Pull request by markumed1**
final

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,16 +1,23 @@
-import { Routes, Route } from "react-router-dom";
-import React from "react";
+import React, { useEffect } from "react";
```
Comment (markumed1) [inline src/App.js:1]: useEffect Hook (Fetching data, Reading from local storage, Registering and deregistering listeners)

Comment (markumed1) [inline src/App.js:1]: useEffect is used for react life cycles: Mounting, Updating, and Unmounting

[React Lifecycle](https://www.w3schools.com/react/react_lifecycle.asp#:~:text=Each%20component%20in%20React%20has,Mounting%2C%20Updating%2C%20and%20Unmounting.)

File: src/App.js [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
+import { Routes, Route, useLocation, useSearchParams } from "react-router-dom";
```
Comment (markumed1) [inline src/App.js:2]: [Routes, Route](https://www.simplilearn.com/tutorials/reactjs-tutorial/routing-in-reactjs#:~:text=The%20Router%20in%20React%20JS,is%20sent%20to%20that%20route.): To supply the browser with an asynchronous URL that corresponds to the data that will show on the web page.

[react-router-dom](https://blog.logrocket.com/react-router-dom-tutorial-examples/#:~:text=React%20Router%20DOM%20enables%20you,of%20the%20app%20and%20platform.): Contains the DOM bindings for react React Router. In other words, the router components for websites.

File: src/App.js [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
 
 import SideNav from "./components/side-nav/SideNav";
 import Header from "./components/header/Header";
 import Home from "./pages/home/Home";
 import Search from "./pages/search/Search";
 import Login from "./pages/login/Login";
-import { useLocation } from "react-router-dom";
```
Comment (markumed1) [inline src/App.js:9]: From lines 4-9, the purpose of 'import' is to allow using contents from another file and will help write modular code, i.e., splitting code into multiple files.

[Importing in React JS](https://www.educative.io/answers/what-is-importing-and-exporting-in-react-js)

File: src/assets/icons/CreatePlaylistIcon.jsx [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
+      className="Svg-ytk21e-0 jAKAlG"
+    >
+      <path d="M15.25 8a.75.75 0 01-.75.75H8.75v5.75a.75.75 0 01-1.5 0V8.75H1.5a.75.75 0 010-1.5h5.75V1.5a.75.75 0 011.5 0v5.75h5.75a.75.75 0 01.75.75z"></path>
+    </svg>
+  );
+};
 
-export default CreatePlaylistIcon;
\ No newline at end of file
+export default CreatePlaylistIcon;
```
Comment (markumed1) [inline src/assets/icons/CreatePlaylistIcon.jsx:18]: "Exporting" makes the file contents eligible for import.

File: src/assets/icons/CreatePlaylistIcon.jsx [hunk 5]
```diff
@@ -1,9 +1,18 @@
 import React from "react";
 
 const CreatePlaylistIcon = () => {
-    return (
-        <svg role="img" height="10" width="10" aria-hidden="true" viewBox="0 0 16 16" class="Svg-ytk21e-0 jAKAlG"><path d="M15.25 8a.75.75 0 01-.75.75H8.75v5.75a.75.75 0 01-1.5 0V8.75H1.5a.75.75 0 010-1.5h5.75V1.5a.75.75 0 011.5 0v5.75h5.75a.75.75 0 01.75.75z"></path></svg>
-    );
-}
+  return (
```
Comment (markumed1) [inline src/assets/icons/CreatePlaylistIcon.jsx:4]: the 'return' method in React is a way to return data from a component. It returns the data that was passed into the component, which the parent component can then use. The 'return' methos is generally used when you want to return a single value from a component.

[return method](https://blog.geekster.in/difference-between-render-and-return-method-in-reactjs/#:~:text=The%20return%20method%20in%20React,single%20value%20from%20a%20component.)

File: src/App.js [hunk 6]
```diff
@@ -1,16 +1,23 @@
...
 
 function App() {
```
Comment (markumed1) [inline src/App.js:10]: 'functional' component is just a simple JS function; it accepts the data in the form of props and returns the react element. Whereas the class component will be created using the class keyword, and it extends the React.

File: src/App.js [hunk 7]
```diff
@@ -1,16 +1,23 @@
...
   const location = useLocation();
   const shouldShowHeaderAndSidenav = location.pathname !== "/login";
+  const [searchParams] = useSearchParams();
+
```
Comment (markumed1) [inline src/App.js:14]: lines 11-13 and 16 the word 'const' is short for contants, which indicates the variables value won't change. It creates block-scoped constants, much like variables declared using the **let** keyword.

File: src/App.js [hunk 8]
```diff
@@ -1,16 +1,23 @@
...
+  useEffect(() => {
```
Comment (markumed1) [inline src/App.js:15]: the 'useEffect' hook, which listens for changes, is a hook that executes a block of code when something changes or a function is triggers. Essentially, the useEffect' hook lets you perform side effect functional components  when the application mounts or when state or prop gets updated. 

[useEffect Hook](https://www.educative.io/answers/how-to-use-the-reactjs-useeffect-hook?utm_campaign=brand_educative&utm_source=google&utm_medium=ppc&utm_content=performance_max&eid=5082902844932096&utm_term=&utm_campaign=%5BNew%5D+Performance+Max&utm_source=adwords&utm_medium=ppc&hsa_acc=5451446008&hsa_cam=18511913007&hsa_grp=&hsa_ad=&hsa_src=x&hsa_tgt=&hsa_kw=&hsa_mt=&hsa_net=adwords&hsa_ver=3&gclid=CjwKCAjw5dqgBhBNEiwA7PryaMbZG3iycyR8y6lKztGr5Vge3Ed4BQM_Gq_OQqVoAIRIEbZkmKlOURoC1JEQAvD_BwE)

File: src/App.js [hunk 9]
```diff
@@ -1,16 +1,23 @@
...
+    const tokenInUri = searchParams.get("code");
+    if (tokenInUri) {
```
Comment (markumed1) [inline src/App.js:17]: use 'if' to specify a block of code to be executed, if a specified condition is true.

File: src/assets/icons/CreatePlaylistIcon.jsx [hunk 10]
```diff
@@ -1,9 +1,18 @@
...
+    <svg
```
Comment (markumed1) [inline src/assets/icons/CreatePlaylistIcon.jsx:5]: Scalable Vector Graphics. It is used to define vector-based graphics for the Web. Every element and every attribute in SVG files can be animated.

**Review by kenttk — APPROVED** [review]
let's merge em!

**Pull request merged**